doc/reference/common-weaknesses/confused-instance.rst

```diff
@@ -0,0 +1,15 @@
+Separate contract instances may be confused
+===========================================
+
+When a number of on-chain contract instances are started by creating a number of UTXOs addressed to the corresponding script. If there is no unique parameter in the script, then different off-chain instances may address their UTXOs to the same script address, which can lead to confusion and incorrect behavior. (This happens, for example, when the state machine library is used without ‘thread tokens’).
```

Comment:
While I like "contract instance" as a piece of jargon, it's sadly not widely used. Not sure if we should keep using it or try and replace it or expand it.